### DISCUSSION TITLE
Compiling looks still for old versions.

### DISCUSSION THREAD

**Discussion by stefonarch** [lxqt/lxqt#2235] [Q&A] [answered]
It's some time (after switching to makepkg from direct install) that on one machine I stumble open those.
 Latest for `qtxdg-tools` 3.9.1:
```
CMake Error at /usr/share/cmake/qt5xdgiconloader/qt5xdgiconloader-targets.cmake:79 (message):
  The imported target "Qt5XdgIconLoader" references the file                                                     
                                                                                                                 
     "/usr/lib/libQt5XdgIconLoader.so.3.9.0"      `
```

It happened also looking for 
```
/usr/lib/liblxqt-globalkeys-ui.so.1.0.
/usr/lib/liblxqt-globalkeys.so.1.0.0
/usr/lib/libqtermwidget5.so.1.0.0
/usr/lib/libfm-qt.so.10.0.0

```
I temporary solved using symlinks as only versioning changed  but I like to solve for good. I ckecked `/usr/share/cmake` for older stuff but nothing.

**Comment by tsujan** [comment, marked as answer]
That just means you haven't upgraded `libqtxdg` before compiling `qtxdg-tools`-3.9.1; have you?

**Reply by stefonarch** [reply]
Sure I did, and it doesn't explain that it happened with many others too...
```
yay -Qi libqtxdg-git 
Nome                      : libqtxdg-git
Versione                  : 3.9.1-1

```

**Reply by stefonarch** [reply]
For some strange reason the only one of the symlinks still needed atm is  `/usr/lib/libqtermwidget5.so.1.0.0 → /usr/lib/libqtermwidget5.so.1.1.0`.
It looks like a  loop to me.

**Reply by tsujan** [reply]
So, I haven't understood the problem at all.

I use `makepkg` with my own PKGBUILDs — no AUR helper.

**Reply by stefonarch** [reply]
Me too, and it looks as I found  now the issue just sitting in front of the screen - my script to compile, install and save packages was buggy and installed the previous package which usually was just the prerelease... shame on me. 

But thanks for your hint, I found it out installing the latest package with `yay -U libqtxdg-git-3.9.1-1-x86_64.pkg.tar`

**Reply by stefonarch** [reply]
That explains also why at some point it installed the last release package.

**Reply by stefonarch** [reply]
No end to errors, for some reason the last step fails (PKGBUILD adapted from @yan12125 )
```
  cd "$srcdir/$_pkgname"/pyqt
  CXXFLAGS="-I$PWD/../lib -I$PWD/../../build/lib" LDFLAGS="-L$PWD/../../build" sip-wheel --verbose

```
```

[100%] Built target qtermwidget5
Traceback (most recent call last):
  File "/usr/bin/sip-wheel", line 33, in <module>
    sys.exit(load_entry_point('sip==6.6.1', 'console_scripts', 'sip-wheel')())
  File "/usr/bin/sip-wheel", line 25, in importlib_load_entry_point
    return next(matches).load()
StopIteration
==> ERRORE: Si è verificato un errore in build().
```

**Reply by tsujan** [reply]
Let me confess! To have the most flexible installation, I do "terrible" things that no Arch user should do ;) For example, my `PKGBUILD` files don't have `source` in them; I just put them inside the extracted sources and issue `makepkg`.

Other "bad" things that I do: `depends` and `makedepends` are empty and `pkgver` is the date.

When I was a Debian user, I did worse things. I have my reasons for that. Have never encountered a problem :P
